cloud/kubernetes/helm/yugabyte/templates/service.yaml

```diff
@@ -196,6 +197,7 @@ spec:
           - "--tserver_master_addrs={{range $index := until (int ($root.Values.replicas.master))}}{{if ne $index 0}},{{end}}yb-master-{{ $index }}.yb-masters.$(NAMESPACE):7100{{end}}"
           - "--tserver_master_replication_factor={{ $root.Values.replicas.master }}"
           - "--metric_node_name=$(HOSTNAME)"
+          - "--memory_limit_hard_bytes={{ $root.Values.resource.tserver.limits.memory | regexFind "\\d+" | mul 1024 | mul 1024 | mul 870 }}"
```

Comment:
@Arnav15 so what i mentioned was to move and create a template function inside _helpers.tpl

```
{{/*
Derive the memory hard limit for each POD based on the memory limit.
Since the memory is represented in <x>GBi, we use this function to convert that into bytes.
*/}}
{{- define "yugabyte.memory_hard_limit" -}}
{{- printf "%d" .limits.memory | regexFind "\\d+" | mul 1024 | mul 1024 | mul 870 }}
{{- end -}}
```

Once you defined this, you could just inline it like this
`- "--memory_limit_hard_bytes={{ template "yugabyte.memory_hard_limit" $root.Values.resource.tserver }}"`

Reply:
Done!